protocol/answer/answer.json

```
@@ -1,65 +1,59 @@
 {
-	"$schema": "http://json-schema.org/draft-07/schema#",
-	"$id": "https://raw.githubusercontent.com/dedis/student_21_pop/master/protocol/answer/answer.json",
-	"title": "Match a custom JsonRpc 2.0 message answer",
-	"description": "Match a positive or negative server answer",
-	"type": "object",
-	"properties": {
-		"jsonrpc": {
-			"description": "[String] JsonRpc version",
-			"const": "2.0",
-			"$comment": "should always be \"2.0\""
-		},
-		"result": {
-			"description": "In case of positive answer, result of the client query",
-			"oneOf": [
-				{
-					"$ref": "positiveAnswer/positiveAnswerGeneral.json"
-				},
-				{
-					"$ref": "positiveAnswer/positiveAnswerCatchup.json"
-				}
-			],
-			"$comment": "Note: this field is absent if the answer is negative"
-		},
-		"error": {
-			"description": "In case of negative answer, error generated by the client query",
-			"$ref": "negativeAnswer/error.json"
-		},
-		"id": {
-			"oneOf": [
-				{
-					"type": "integer"
-				},
-				{
-					"type": "null"
-				}
-			],
-			"$comment": "The id match the request id. If there was an error in detecting the id, it must be null"
-		}
-	},
-	"additionalProperties": false,
-	"allOf": [
-		{
-			"required": [
-				"jsonrpc",
-				"id"
-			]
-		},
-		{
-			"oneOf": [
-				{
-					"required": [
-						"result"
-					]
-				},
-				{
-					"required": [
-						"error"
-					]
-				}
-			]
-		}
-	],
-	"$comment": "Note: required fields: (\"jsonrpc\" \\and (\"result\" \\or \"error\") \\and \"id\")"
-}
\ No newline at end of file
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/dedis/student_21_pop/master/protocol/answer/answer.json",
+    "title": "Match a custom JsonRpc 2.0 message answer",
+    "description": "Match a positive or negative server answer",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+        "result": {
+            "description": "In case of positive answer, result of the client query",
+            "oneOf": [
+                {
+                    "type": "integer",
+                    "const": 0
+                },
+                {
+                    "type": "array",
+                    "items": {
+                        "$ref": "../query/method/message/message.json"
+                    },
+                    "minItems": 1,
+                    "$comment": "if there are no items, then `0` must be returned"
```

Warning: this is a change of spec, and probably a wrong one too.
It doesn't make sense to change the return type based on whether the array is empty or not.

```suggestion
                    "$minItems": 0,
                    "$comment": "Return value for a `catchup` request"
```